fatal error: DOtherSide.moc: No such file or directory

CI Linux builds have been failing on `master` since May 14th([#938](https://ci.status.im/job/status-desktop/job/branches/job/linux/job/master/938/)) on this error:
```
cd .../vendor/DOtherSide/build/lib && /usr/bin/c++   -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_DEBUG -DQT_QMLMODELS_LIB -DQT_QML_LIB -DQT_QUICKCONTROLS2_LIB -DQT_QUICK_LIB -DQT_WIDGETS_LIB -I/home/jenkins/workspace/s-desktop_branches_linux_PR-2315/vendor/DOtherSide/lib/include -I/home/jenkins/workspace/s-desktop_branches_linux_PR-2315/vendor/DOtherSide/lib/include/Qt -isystem /opt/qt/5.14.0/gcc_64/include -isystem /opt/qt/5.14.0/gcc_64/include/QtCore -isystem /opt/qt/5.14.0/gcc_64/./mkspecs/linux-g++ -isystem /opt/qt/5.14.0/gcc_64/include/QtGui -isystem /opt/qt/5.14.0/gcc_64/include/QtWidgets -isystem /opt/qt/5.14.0/gcc_64/include/QtQml -isystem /opt/qt/5.14.0/gcc_64/include/QtNetwork -isystem /opt/qt/5.14.0/gcc_64/include/QtQuick -isystem /opt/qt/5.14.0/gcc_64/include/QtQmlModels -isystem /opt/qt/5.14.0/gcc_64/include/QtQuickControls2  -O3 -DNDEBUG   -Wall -Wno-long-long -pedantic -fPIC -std=gnu++11 -o CMakeFiles/DOtherSideStatic.dir/src/DosQMetaObject.cpp.o -c /home/jenkins/workspace/s-desktop_branches_linux_PR-2315/vendor/DOtherSide/lib/src/DosQMetaObject.cpp
.../vendor/DOtherSide/lib/src/DOtherSide.cpp:1322:26: fatal error: DOtherSide.moc: No such file or directory
```
I've purged all hosts workspaces and restarted them, so the issue appears to be with the repo itself.

I tried the older `1.0.1` image and it failed with the same error:
```
23:15:08  make: *** [vendor/DOtherSide/build/lib/libDOtherSideStatic.a] Error 2
```
https://ci.status.im/job/status-desktop/job/branches/job/linux/job/master/950/

So it's definitely not the image.

--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -20,6 +20,7 @@ macro(add_target name type)
         include/DOtherSide/DosIQAbstractItemModelImpl.h
         include/DOtherSide/DosQAbstractItemModel.h
         include/DOtherSide/Utils.h
+        include/DOtherSide/DosDockClicker.h
         src/DOtherSide.cpp
         src/DosQMetaObject.cpp
         src/DosQDeclarative.cpp
@@ -28,6 +29,7 @@ macro(add_target name type)
         src/DosQObjectImpl.cpp
         src/DosQAbstractItemModel.cpp
         src/DosQQuickImageProvider.cpp
+        src/DosDockClicker.cpp
     )
 
     if (WIN32)

--- a/lib/include/DOtherSide/DosDockClicker.h
+++ b/lib/include/DOtherSide/DosDockClicker.h
@@ -1,0 +1,27 @@
+#ifndef DOCKCLICKER_H
+#define DOCKCLICKER_H
+
+#include <QObject>
+
+#include <QQmlApplicationEngine>
+#include "DOtherSidetypes.h"
+
+class DockClicker : public QObject
+{
+    Q_OBJECT
+
+private:
+    Qt::ApplicationState _prevAppState;
+    QQmlApplicationEngine* _engine;
+
+protected:
+    bool eventFilter(QObject *obj, QEvent *event) override;
+
+public:
+    DockClicker(DosQQmlApplicationEngine *vptr) {
+        auto engine = static_cast<QQmlApplicationEngine *>(vptr);
+        _engine = engine;
+    }
+};
+
+#endif // DOCKCLICKER_H

--- a/lib/src/DOtherSide.cpp
+++ b/lib/src/DOtherSide.cpp
@@ -57,6 +57,7 @@
 #include "DOtherSide/DosQAbstractItemModel.h"
 #include "DOtherSide/DosQDeclarative.h"
 #include "DOtherSide/DosQQuickImageProvider.h"
+#include "DOtherSide/DosDockClicker.h"
 
 namespace {
 
@@ -153,46 +154,6 @@ void dos_qguiapplication_exec()
 void dos_qguiapplication_quit()
 {
     qGuiApp->quit();
-}
-
-class DockClicker : public QObject
-{
-    Q_OBJECT
-
-    private:
-        Qt::ApplicationState _prevAppState;
-        QQmlApplicationEngine* _engine;
-
-    protected:
-        bool eventFilter(QObject *obj, QEvent *event) override;
-
-    public:
-        DockClicker(::DosQQmlApplicationEngine *vptr) {
-            auto engine = static_cast<QQmlApplicationEngine *>(vptr);
-            _engine = engine;
-        }
-};
-
-bool DockClicker::eventFilter(QObject *obj, QEvent *event)
-{
-    #ifdef Q_OS_MACOS
-        if (obj == qApp) {
-            if (event->type() == QEvent::ApplicationStateChange) {
-                auto ev = static_cast<QApplicationStateChangeEvent*>(event);
-                if (_prevAppState == Qt::ApplicationActive && ev->applicationState() == Qt::ApplicationActive) {
-                    QObject *topLevel = _engine->rootObjects().value(0);
-                    QQuickWindow *window = qobject_cast<QQuickWindow *>(topLevel);
-                    window->setVisible(true);
-                    window->showNormal();
-                }
-                _prevAppState = ev->applicationState();
-                return true;
-            } else {
-                return false;
-            }
-        }
-    #endif // Q_OS_MACOS
-        return QObject::eventFilter(obj, event);
 }
 
 void dos_qapplication_create(char *appName)
@@ -1319,4 +1280,3 @@ char *dos_qurl_replaceHostAndAddPath(char* url, char* newScheme, char* newHost, 
 
     return convert_to_cstring(newQurl.toString());
 }
-#include "DOtherSide.moc"

--- a/lib/src/DosDockClicker.cpp
+++ b/lib/src/DosDockClicker.cpp
@@ -1,0 +1,27 @@
+#include "DOtherSide/DosDockClicker.h"
+
+#include <QQuickWindow>
+#include <QCoreApplication>
+
+
+bool DockClicker::eventFilter(QObject *obj, QEvent *event)
+{
+#ifdef Q_OS_MACOS
+    if (obj == qApp) {
+        if (event->type() == QEvent::ApplicationStateChange) {
+            auto ev = static_cast<QApplicationStateChangeEvent*>(event);
+            if (_prevAppState == Qt::ApplicationActive && ev->applicationState() == Qt::ApplicationActive) {
+                QObject *topLevel = _engine->rootObjects().value(0);
+                QQuickWindow *window = qobject_cast<QQuickWindow *>(topLevel);
+                window->setVisible(true);
+                window->showNormal();
+            }
+            _prevAppState = ev->applicationState();
+            return true;
+        } else {
+            return false;
+        }
+    }
+#endif // Q_OS_MACOS
+    return QObject::eventFilter(obj, event);
+}

